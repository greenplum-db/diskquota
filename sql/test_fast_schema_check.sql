-- Test SCHEMA
CREATE SCHEMA s1;
SET search_path to s1;

CREATE TABLE a(i int);
SELECT diskquota.set_schema_quota('s1', '1 MB');
INSERT INTO a SELECT generate_series(1,200000);
SELECT pg_sleep(10);

select a.nspsize_in_bytes = b.nspsize_in_bytes,
	a.quota_in_mb = b.quota_in_mb,
	a.schema_oid = b.schema_oid
from diskquota.show_schema_quota_view as a,
	diskquota.show_schema_quota_proximate_view as b
where b.schema_name = 's1' and a.schema_name = b.schema_name;

RESET search_path;
DROP TABLE s1.a;
DROP SCHEMA s1;

