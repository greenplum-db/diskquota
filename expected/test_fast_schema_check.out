-- Test SCHEMA
CREATE SCHEMA s1;
SET search_path to s1;
CREATE TABLE a(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
SELECT diskquota.set_schema_quota('s1', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO a SELECT generate_series(1,200000);
SELECT pg_sleep(10);
 pg_sleep 
----------
 
(1 row)

select a.nspsize_in_bytes = b.nspsize_in_bytes,
	a.quota_in_mb = b.quota_in_mb,
	a.schema_oid = b.schema_oid
from diskquota.show_schema_quota_view as a,
	diskquota.show_schema_quota_proximate_view as b
where b.schema_name = 's1' and a.schema_name = b.schema_name;
 ?column? | ?column? | ?column? 
----------+----------+----------
 t        | t        | t
(1 row)

RESET search_path;
DROP TABLE s1.a;
DROP SCHEMA s1;
